pkg/aws/actuator/actuator.go

```diff
@@ -545,12 +547,12 @@ func (a *AWSActuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequ
 	}
 
 	// Check if the user has the expected tags:
-	if !userHasExpectedTags(logger, userOut, infraName, string(clusterUUID)) {
+	if !userHasExpectedTags(logger, userOut, string(clusterUUID), infra) {
 		if rootAWSClient == nil {
 			return fmt.Errorf("no root AWS client available, cred secret may not exist: %s/%s", constants.CloudCredSecretNamespace, constants.AWSCloudCredSecretName)
 		}
 
-		err = a.tagUser(logger, rootAWSClient, awsStatus.User, infraName, string(clusterUUID))
+		err = a.tagUser(logger, rootAWSClient, awsStatus.User, infra.Status.InfrastructureName, string(clusterUUID))
```

Comment:
pass the full Infrastructure to tagUser() so that tagUser() doesn't have to fetch it and potentially error due to that reason.

Reply:
Sorry, Missed it. Noted.